### PR TITLE
Email Link Sign In Documentation + Backstack fixes

### PR DESCRIPTION
**Pull request by lsirac**
https://github.com/firebase/FirebaseUI-Android/issues/1501

### CODE REVIEW

File: auth/src/main/java/com/firebase/ui/auth/ui/email/EmailLinkFragment.java [hunk 1]
```diff
@@ -85,7 +89,8 @@ public void onActivityCreated(@Nullable Bundle savedInstanceState) {
         ActionCodeSettings actionCodeSettings = getArguments().getParcelable(ExtraConstants
                 .ACTION_CODE_SETTINGS);
 
-        if (savedInstanceState == null || !savedInstanceState.getBoolean(EMAIL_SENT)) {
+        if (!mEmailSent && (savedInstanceState == null || !savedInstanceState.getBoolean
```
Comment (samtstern) [inline auth/src/main/java/com/firebase/ui/auth/ui/email/EmailLinkFragment.java]: Your crazy formatter is at it again!

File: auth/src/main/java/com/firebase/ui/auth/ui/email/EmailLinkFragment.java [hunk 2]
```diff
@@ -34,6 +34,9 @@
     private TroubleSigningInListener mListener;
     private ScrollView mTopLevelView;
 
+    // Used to avoid sending a new email when popping off the fragment backstack
+    private boolean mEmailSent;
```
Comment (samtstern) [inline auth/src/main/java/com/firebase/ui/auth/ui/email/EmailLinkFragment.java]: Does this survive rotation?

File: auth/src/main/java/com/firebase/ui/auth/ui/email/EmailActivity.java [hunk 3]
```diff
@@ -214,16 +214,22 @@ private void showRegisterEmailLinkFragment(AuthUI.IdpConfig emailConfig,
     }
 
 
-    private void switchFragment(Fragment fragment, String tag, boolean withTransition) {
+    private void switchFragment(Fragment fragment, String tag, boolean withTransition,
```
Comment (samtstern) [inline auth/src/main/java/com/firebase/ui/auth/ui/email/EmailActivity.java]: Can we make an overload without the new param that defaults it to false?

File: auth/src/main/java/com/firebase/ui/auth/ui/email/EmailLinkFragment.java (hunk 1 above)
Comment (lsirac) [reply auth/src/main/java/com/firebase/ui/auth/ui/email/EmailLinkFragment.java]: :cry: fixed!

File: auth/src/main/java/com/firebase/ui/auth/ui/email/EmailLinkFragment.java (hunk 2 above)
Comment (lsirac) [reply auth/src/main/java/com/firebase/ui/auth/ui/email/EmailLinkFragment.java]: Nope, my bad :-1: . I also fixed the case where we go from EmailLinkFragment -> TroubleSigningInFragment -> EmailLinkFragment (by clicking resend), we'd be at EmailLinkFragment and have to press the back twice to get back to the nascar screen.

File: auth/src/main/java/com/firebase/ui/auth/ui/email/EmailActivity.java (hunk 3 above)
Comment (lsirac) [reply auth/src/main/java/com/firebase/ui/auth/ui/email/EmailActivity.java]: We have one already that sets both booleans to false.

File: auth/src/main/java/com/firebase/ui/auth/ui/email/EmailActivity.java [hunk 4]
```diff
@@ -173,11 +173,14 @@ public void onNewUser(User user) {
     public void onTroubleSigningIn(String email) {
         TroubleSigningInFragment troubleSigningInFragment = TroubleSigningInFragment.newInstance
                 (email);
-        switchFragment(troubleSigningInFragment, TroubleSigningInFragment.TAG, true);
+        switchFragment(troubleSigningInFragment, TroubleSigningInFragment.TAG, true, true);
     }
 
     @Override
     public void onClickResendEmail(String email) {
+        if (getSupportFragmentManager().getBackStackEntryCount() > 0) {
```
Comment (samtstern) [inline auth/src/main/java/com/firebase/ui/auth/ui/email/EmailActivity.java]: Can you just add a comment here detailing the assumption you're making?  As I understand it, you're assuming that if there's any back stack at all here it's due to the trouble sending flow and you want to pop from the stack.

Comment (lsirac) [reply auth/src/main/java/com/firebase/ui/auth/ui/email/EmailActivity.java]: Good idea. Done. :smiley:

**Pull request merged**